### PR TITLE
chore(ts): Convert adminWarnings

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx [hunk 1]
```diff
@@ -3,13 +3,16 @@ import React from 'react';
...
+type Data = {
+  groups: [groupName: string, grouppedWarnings: string[]][];
+  warnings: string[];
+};
+
+type State = AsyncView['state'] & {data: Data};
 
-  getEndpoint() {
-    return '/internal/warnings/';
+export default class AdminSettings extends AsyncView<{}, State> {
```
Comment (taylangocmen) [inline src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:13]: Shouldn't this be named `AdminWarnings`, doesn't affect the code but it feels wrong

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx [hunk 2]
```diff
@@ -3,13 +3,16 @@ import React from 'react';
...
+  getEndpoints(): [string, string][] {
```
Comment (taylangocmen) [inline src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:14]: `ReturnType<AsyncView['getEndpoints']>` is the convention we've been use recently

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx [hunk 3]
```diff
@@ -3,13 +3,16 @@ import React from 'react';
...
-export default class AdminSettings extends AsyncView {
-  getEndpoints() {
-    return [['data', this.getEndpoint()]];
-  }
+type Data = {
+  groups: [groupName: string, grouppedWarnings: string[]][];
+  warnings: string[];
+};
+
+type State = AsyncView['state'] & {data: Data};
```
Comment (taylangocmen) [inline src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:11]: Afaiu `AsyncComponent` sets these fetched values to `null` if there is an error, so we would want to account for that. Please correct me if I'm wrong @priscilawebdev 
```suggestion
type State = AsyncView['state'] & {data: Data | null};
```

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:13]: Yes it should lol

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:11]: that is right @taylangocmen 😉

**Comment by priscilawebdev** [comment]
with @taylangocmen suggestions looks good 😉

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/admin/adminWarnings.tsx:14]: gonna batch fix these in a sec 👍

File: src/sentry/static/sentry/app/views/admin/adminWarnings.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/admin/adminWarnings.tsx]: I give in 🤐

**Comment by EvanPurkhiser** [comment]
fixed up

**Review by matejminar — APPROVED** [review]

**Pull request merged**